src/proto/h1/dispatch.rs

```diff
@@ -268,19 +268,22 @@ where
                 if let Some(msg) = ready!(self.dispatch.poll_msg(cx)) {
                     let (head, mut body) = msg.map_err(crate::Error::new_user_service)?;
 
-                    // Check if the body knows its full data immediately.
-                    //
-                    // If so, we can skip a bit of bookkeeping that streaming
-                    // bodies need to do.
-                    if let Some(full) = body.__hyper_full_data(FullDataArg(())).0 {
-                        self.conn.write_full_msg(head, full);
-                        return Poll::Ready(Ok(()));
+
+                    let hint = body.size_hint();
+                    if hint.upper() == Some(hint.lower()) {
+                        if let (Some(mut body), _clear_body) = OptGuard::new(self.body_rx.as_mut()).guard_mut() {
```

Comment:
I believe this should replicate what `__hyper_full_data` did previously.

Reply:
It's actually a little different, so I suggest just removing that optimization for now, as I have a simple idea to get it back.